pkg/util/hashs/factory.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hashs
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"github.com/cespare/xxhash/v2"
+	"hash"
```

Comment:
Why need standard hash? IMO you can define a new Hash interface, which has Write and Sum methods

Reply:
referenced the `hash.Hash` interface.

In my opinion, the factory should create an instance that implements the `hash.Hash` interface, so that it is convenient to replace the implementation later. When there is a better implementation than `xxhash`, we can directly replace the implementation in the `New` method without modifying other places.